src/crypto-providers/signingValidation.ts

```diff
@@ -0,0 +1,256 @@
+    throw Error(Errors.TooManyPoolColdSigningFilesError)
+  }
+  if (!params.unsignedTxParsed.mint && mintSigningFiles.length > 0) {
+    throw Error(Errors.TooManyMintSigningFilesError)
+  }
+  if (multisigSigningFiles.length > 0) {
+    throw Error(Errors.TooManyMultisigSigningFilesError)
+  }
+}
+
+const validateOrdinarySigning = (params: SigningParameters) => {
+  validateOrdinaryWitnesses(params)
+  // these checks should be performed only with createSignedTx call (when the signing file set is
+  // expected to be complete) on top of validateOrdinaryWitnesses
+
+  const {
+    paymentSigningFiles, stakeSigningFiles, poolColdSigningFiles,
+  } = filterSigningFiles(params.hwSigningFileData)
+
+  if (paymentSigningFiles.length === 0) {
+    throw Error(Errors.MissingPaymentSigningFileError)
+  }
+
+  const {
+    numStakeOtherItems, numPoolColdItems,
+  } = _countWitnessableItems(params.unsignedTxParsed)
+
+  if (numStakeOtherItems > 0 && (stakeSigningFiles.length === 0)) {
+    throw Error(Errors.MissingStakeSigningFileError)
+  }
+  if (numPoolColdItems > 0 && (poolColdSigningFiles.length === 0)) {
+    throw Error(Errors.MissingPoolColdSigningFileError)
+  }
+}
+
+const validatePoolRegistrationTx = (unsignedTxParsed: _UnsignedTxParsed) => {
+  _validateStakeCredentialsAreAllOfType(
+    unsignedTxParsed, StakeCredentialType.ADDR_KEY_HASH, Errors.ScriptStakeCredentialInOrdinaryTx,
+  )
+
+  if (unsignedTxParsed.certificates.length !== 1) {
+    throw Error(Errors.MultipleCertificatesWithPoolRegError)
+  }
+  if (unsignedTxParsed.withdrawals.length > 0) {
+    throw Error(Errors.WithdrawalIncludedWithPoolRegError)
+  }
+}
+
+const validatePoolOwnerWitnesses = (params: SigningParameters) => {
+  const {
+    paymentSigningFiles, stakeSigningFiles, poolColdSigningFiles, mintSigningFiles, multisigSigningFiles,
+  } = filterSigningFiles(params.hwSigningFileData)
+
+  if (paymentSigningFiles.length > 0) {
+    throw Error(Errors.TooManyPaymentFilesWithPoolRegError)
+  }
+  if (stakeSigningFiles.length === 0) {
+    throw Error(Errors.MissingStakeSigningFileError)
+  }
+  if (stakeSigningFiles.length > 1) {
+    // we need exactly one signing file in order to unambiguously determine the owner to be witnessed
+    throw Error(Errors.TooManyStakeSigningFilesError)
+  }
+  if (poolColdSigningFiles.length > 0) {
+    throw Error(Errors.TooManyPoolColdSigningFilesError)
+  }
+  if (mintSigningFiles.length > 0) {
+    throw Error(Errors.TooManyMintSigningFilesError)
+  }
+  if (multisigSigningFiles.length > 0) {
+    throw Error(Errors.TooManyMultisigSigningFilesError)
+  }
+}
+
+const validatePoolOperatorWitnesses = (params: SigningParameters) => {
+  const {
+    stakeSigningFiles, poolColdSigningFiles, multisigSigningFiles,
+  } = filterSigningFiles(params.hwSigningFileData)
+
+  if (stakeSigningFiles.length > 0) {
+    throw Error(Errors.TooManyStakeSigningFilesError)
+  }
+  if (poolColdSigningFiles.length === 0) {
```

Comment:
I'd consider enforcing a single pool cold signing file. Pool registrations are done  very rarely and it will force the user to make sure he knows what he is doing.

Reply:
That makes sense, changed.